cmd/process-agent/collector.go

```diff
@@ -258,3 +289,63 @@ func getContainerCount(mb model.MessageBody) int {
 	}
 	return 0
 }
+
+func readResponseStatuses(responses chan forwarder.Response) []*model.CollectorStatus {
+	var statuses []*model.CollectorStatus
+
+	for response := range responses {
+		if response.Err != nil {
+			log.Errorf("Error from %s: %s", response.Domain, response.Err)
+			continue
+		}
+
+		if response.StatusCode >= 300 {
+			log.Errorf("Invalid response from %s: %d -> %s", response.Domain, response.StatusCode, response.Err)
+			continue
+		}
```

Comment:
Are we 100% sure that 3xx / redirect code should be considered as invalid ?

Reply:
Yes, this would normally be handled inside the `net/http`.  Anything that makes its way back to the caller that isn't 2xx is invalid.